src/acquisition/afhsb/afhsb_csv.py

```diff
@@ -19,174 +19,186 @@
-		return year, week_num
-
-	results_dict = {}
-	for _, row in enumerate(reader):
-		# if (r >= 1000000): break
-		if get_field(row, 'type') != "Outpt":
-			continue
-		year, week_num = row2epiweek(row)
-		dmisid = get_field(row, 'DMISID')
-		flu_cat = row2flu(row)
-
-		key_list = [year, week_num, dmisid, flu_cat]
-		curr_dict = results_dict
-		for i, key in enumerate(key_list):
-			if i == len(key_list) - 1:
-				if key not in curr_dict:
-					curr_dict[key] = 0
-				curr_dict[key] += 1
-			else:
-				if key not in curr_dict:
-					curr_dict[key] = {}
-				curr_dict = curr_dict[key]
-
-	results_path = os.path.join(TARGET_DIR, targetfile)
-	with open(results_path, 'wb') as f:
-		pickle.dump(results_dict, f, pickle.HIGHEST_PROTOCOL)
+    reader = sas7bdat.SAS7BDAT(os.path.join(SOURCE_DIR, sourcefile), skip_header=True)
+    # map column names to column indices
+    col_2_idx = {column.name.decode("utf-8"): column.col_id for column in reader.columns}
+
+    def get_field(row, column):
+        return row[col_2_idx[column]]
+
+    def row2flu(row):
+        for i in range(1, 9):
+            dx = get_field(row, "dx{}".format(i))
+            flu_cat = get_flu_cat(dx)
+            if flu_cat is not None:
+                return flu_cat
+        return 0
+
+    def row2epiweek(row):
+        date = get_field(row, "d_event")
+        year, month, day = date.year, date.month, date.day
+        week_tuple = epi.Week.fromdate(year, month, day).weektuple()
+        year, week_num = week_tuple[0], week_tuple[1]
+        return year, week_num
+
+    results_dict = {}
+    for _, row in enumerate(reader):
+        # if (r >= 1000000): break
+        if get_field(row, "type") != "Outpt":
+            continue
+        year, week_num = row2epiweek(row)
+        dmisid = get_field(row, "DMISID")
+        flu_cat = row2flu(row)
+
+        key_list = [year, week_num, dmisid, flu_cat]
+        curr_dict = results_dict
+        for i, key in enumerate(key_list):
+            if i == len(key_list) - 1:
+                if key not in curr_dict:
+                    curr_dict[key] = 0
+                curr_dict[key] += 1
+            else:
+                if key not in curr_dict:
+                    curr_dict[key] = {}
+                curr_dict = curr_dict[key]
+
+    results_path = os.path.join(TARGET_DIR, targetfile)
+    with open(results_path, "wb") as f:
+        pickle.dump(results_dict, f, pickle.HIGHEST_PROTOCOL)
 
 
 ################# Functions for geographical information ####################
 
+
 def get_country_mapping():
-	filename = "country_codes.csv"
-	mapping = dict()
-	with open(os.path.join(TARGET_DIR, filename), "r") as csvfile:
-		reader = csv.DictReader(csvfile)
-		for row in reader:
-			print(row.keys())
-			alpha2 = row['alpha-2']
-			alpha3 = row['alpha-3']
-			mapping[alpha2] = alpha3
-
-	return mapping
+    filename = "country_codes.csv"
+    mapping = dict()
+    with open(os.path.join(TARGET_DIR, filename)) as csvfile:
```

Comment:
FWIW, I also used [`pyupgrade`](https://github.com/asottile/pyupgrade) here, which made a few changes like this

Reply:
i like the pyupgrade changes